reactor-netty-http/src/test/java/reactor/netty/http/Http2Tests.java

```diff
@@ -414,4 +415,34 @@ public void testHttp2ForMemoryLeaks() throws Exception {
 		}
 		System.gc();
 	}
+
+	@Test
+	public void testHttpClientDefaultSslProvider() {
+		HttpClient client = HttpClient.create()
+		                              .wiretap(true);
+
+		doTestHttpClientDefaultSslProvider(client.protocol(HttpProtocol.H2));
+		doTestHttpClientDefaultSslProvider(client.protocol(HttpProtocol.H2)
+		                               .secure());
+		doTestHttpClientDefaultSslProvider(client.secure()
+		                               .protocol(HttpProtocol.H2));
+		doTestHttpClientDefaultSslProvider(client.protocol(HttpProtocol.HTTP11)
+		                               .secure()
+		                               .protocol(HttpProtocol.H2));
```

Comment:
Is this intended?

Reply:
Yes checking that applying a protocol configuration multiple times sets the correct `SslProvider`.
Let's say we have this:

```
HttpClient client1 =
    HttpClient....
           .protocol(HttpProtocol.HTTP11)
	   .secure();
HttpClient client2 = client1.protocol(HttpProtocol.H2)
```

`client1` should run with the default `SslProvider` for `HTTP/1.1`
`client2` for this for `HTTP/2`